docs/frontend-development/css-guidelines.md

```diff
@@ -0,0 +1,399 @@
+```
+
+Compiled CSS is output to the `dist/css/` directory during the build process.
+
+### Import Order
+
+CSS imports with PostCSS are typically organized in the following order:
+
+```css
+/* Example import order */
+/* Globals */
+@import 'globals/variables.css';
+@import 'globals/typography.css';
+@import 'globals/global.css';
+
+/* Mixins */
+@import 'mixins/responsive.css';
+@import 'mixins/utilities.css';
+
+/* Base styles */
+@import 'frontend/base/reset.css';
+@import 'frontend/base/layout.css';
+
+/* Components */
+@import 'frontend/components/buttons.css';
+@import 'frontend/components/forms.css';
+@import 'frontend/components/navigation.css';
+
+/* Templates/Pages */
+@import 'frontend/templates/home.css';
+@import 'frontend/templates/single.css';
+
+/* Blocks */
+@import 'blocks/core-blocks.css';
+@import 'blocks/custom-blocks.css';
+```
+
+## Naming Conventions
+
+### BEM Methodology
+
+The project follows the BEM (Block, Element, Modifier) naming convention:
+
+```css
+.block {}
+.block__element {}
+.block--modifier {}
+.block__element--modifier {}
+```
+
+Examples:
+
+```css
+.card {}
+.card__title {}
+.card__image {}
+.card--featured {}
+.card__title--large {}
+```
+
+### Prefixing
+
+Use a project-specific prefix for global components to avoid conflicts:
+
+```css
+.scaffold-button {}
+.scaffold-form {}
+```
+
+### State Classes
+
+Use `is-` or `has-` prefixes for state classes:
+
+```css
+.is-active {}
+.is-disabled {}
+.has-error {}
+```
+
+## PostCSS Features
+
+### PostCSS Plugins
+
+The WP Scaffold project uses several PostCSS plugins to enhance CSS capabilities:
+
+```js
+// Example PostCSS configuration
+module.exports = {
+  plugins: [
+    require('postcss-import'),
+    require('postcss-preset-env')({
+      stage: 2,
+      features: {
+        'custom-properties': true,
+        'nesting-rules': true
+      }
+    }),
+    require('autoprefixer'),
+    require('cssnano')
+  ]
+};
+```
+
+### Custom Properties (CSS Variables)
+
+Use CSS custom properties for global variables:
+
+```css
+/* Example custom properties */
+:root {
+  /* Colors */
+  --color-primary: #007bff;
+  --color-secondary: #6c757d;
+  --color-success: #28a745;
+  --color-error: #dc3545;
+
+  /* Typography */
+  --font-family-base: 'Helvetica Neue', Arial, sans-serif;
+  --font-size-base: 16px;
+  --line-height-base: 1.5;
+
+  /* Spacing */
+  --spacing-unit: 8px;
+  --spacing-small: var(--spacing-unit);
+  --spacing-medium: calc(var(--spacing-unit) * 2);
+  --spacing-large: calc(var(--spacing-unit) * 3);
+
+  /* Breakpoints */
+  --breakpoint-small: 576px;
+  --breakpoint-medium: 768px;
+  --breakpoint-large: 992px;
+  --breakpoint-xlarge: 1200px;
+}
+```
+
+### CSS Nesting
+
+Use CSS nesting for more maintainable selectors:
+
+```css
+/* Example CSS nesting */
+.card {
+  border: 1px solid #ddd;
+  border-radius: 4px;
+
+  & .card__header {
+    padding: 15px;
+    border-bottom: 1px solid #ddd;
+  }
+
+  & .card__body {
+    padding: 15px;
+  }
```

Comment:
I would say this here is exactly what we shouldn't do because it creates overly specific selectors 🤔